Source/OktaAuthSdk.swift

```diff
@@ -17,10 +17,11 @@ public class OktaAuthSdk {
     public class func authenticate(with url: URL,
                                    username: String,
                                    password: String?,
+                                   httpClient: OktaAuthHTTPClient? = nil,
                                    onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
                                    onError: @escaping (_ error: OktaError) -> Void) {
         
-        let unauthenticatedStatus = OktaAuthStatusUnauthenticated(oktaDomain: url)
+        let unauthenticatedStatus = OktaAuthStatusUnauthenticated(oktaDomain: url, httpClient: httpClient)
```

Comment:
Same as above. Http client should be injected via property injection. So if dev needs OktaAuthHTTPClient, the he/she will do the following:
```
let unauthenticatedStatus = OktaAuthStatusUnauthenticated(oktaDomain: url)
unauthenticatedStatus.restApi.httpsClient = HTTPClient()
unauthenticatedStatus.authenticate(...)
```